ocs_ci/ocs/resources/pod.py

```diff
@@ -887,18 +887,19 @@ def _get_tools_pod_objs():
             cluster_kubeconfig=cluster_kubeconfig,
         )
         running_ct_pods = list()
-        for pod in ct_pod_items:
-            pod_status = ocp_pod_obj.get_resource_status(
-                pod.get("metadata").get("name")
-            )
-            logger.info(f"Pod name: {pod.get('metadata').get('name')}")
-            logger.info(f"Pod status: {pod_status}")
-            if pod_status == constants.STATUS_RUNNING:
-                running_ct_pods.append(pod)
-
-        if not running_ct_pods:
-            raise NoRunningCephToolBoxException
-        return running_ct_pods
+        with config.RunWithProviderConfigContextIfAvailable():
+            for pod in ct_pod_items:
+                pod_status = ocp_pod_obj.get_resource_status(
+                    pod.get("metadata").get("name")
+                )
+                logger.info(f"Pod name: {pod.get('metadata').get('name')}")
+                logger.info(f"Pod status: {pod_status}")
+                if pod_status == constants.STATUS_RUNNING:
+                    running_ct_pods.append(pod)
+
+            if not running_ct_pods:
+                raise NoRunningCephToolBoxException
+            return running_ct_pods
```

Comment:
Please exclude the return statement from the Provider codeblock. The provider context should be only for provider specific actions like work with ceph tools pod.

Reply:
moved the return statement out of provider context

Reply:
The return statement is still part of the code block

Reply:
should be good now